Source/WebKit/UIProcess/wpe/SystemSettingsManagerProxyWPE.cpp

```diff
@@ -0,0 +1,146 @@
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "SystemSettingsManagerProxy.h"
+
+#if ENABLE(WPE_PLATFORM)
+
+#include <vector>
+#include <wpe/WPEDisplay.h>
+#include <wpe/WPESettings.h>
+#include <wtf/glib/GUniquePtr.h>
+
+
+namespace WebKit {
+
+#define SETTING_FONT_NAME "/wpe-platform/font-name"
+#define SETTING_DARK_MODE "/wpe-platform/dark-mode"
+#define SETTING_DISABLE_ANIMATIONS "/wpe-platform/disable-animations"
+
+struct Setting {
+    const char* key;
+    const GVariantType* type;
+    GVariant* defaultValue;
+};
+
+static const std::vector<Setting> defaultSettings = {
+    { SETTING_FONT_NAME, G_VARIANT_TYPE_STRING, g_variant_new_string("Sans 10") },
+    { SETTING_DARK_MODE, G_VARIANT_TYPE_BOOLEAN, g_variant_new_boolean(false) },
+    { SETTING_DISABLE_ANIMATIONS, G_VARIANT_TYPE_BOOLEAN, g_variant_new_boolean(false) },
+};
+
+String SystemSettingsManagerProxy::themeName() const
+{
+    return emptyString();
+}
+
+bool SystemSettingsManagerProxy::darkMode() const
+{
+    return wpe_settings_get_boolean(m_settings, SETTING_DARK_MODE, nullptr);
+}
+
+String SystemSettingsManagerProxy::fontName() const
+{
+    return String::fromUTF8(wpe_settings_get_string(m_settings, SETTING_FONT_NAME, nullptr));
+}
+
+int SystemSettingsManagerProxy::xftAntialias() const
+{
+    return -1;
+}
+
+int SystemSettingsManagerProxy::xftHinting() const
+{
+    return -1;
+}
+
+String SystemSettingsManagerProxy::xftHintStyle() const
+{
+    return emptyString();
+}
+
+String SystemSettingsManagerProxy::xftRGBA() const
+{
+    return "rgb"_s;
+}
+
+int SystemSettingsManagerProxy::xftDPI() const
+{
+    return -1;
+}
+
+bool SystemSettingsManagerProxy::followFontSystemSettings() const
+{
+    return false;
+}
+
+bool SystemSettingsManagerProxy::cursorBlink() const
+{
+    return true;
+}
+
+int SystemSettingsManagerProxy::cursorBlinkTime() const
+{
+    return 1200;
+}
+
+bool SystemSettingsManagerProxy::primaryButtonWarpsSlider() const
+{
+    return true;
+}
+
+bool SystemSettingsManagerProxy::overlayScrolling() const
+{
+    return true;
+}
+
+bool SystemSettingsManagerProxy::enableAnimations() const
+{
+    return !wpe_settings_get_boolean(m_settings, SETTING_DISABLE_ANIMATIONS, nullptr);
+}
+
+SystemSettingsManagerProxy::SystemSettingsManagerProxy()
+    : m_settings(wpe_display_get_settings(wpe_display_get_default()))
+{
+    ASSERT(m_settings);
+
+    for (auto& setting : defaultSettings) {
+        GUniqueOutPtr<GError> error;
+        if (!wpe_settings_register(m_settings, setting.key, setting.type, setting.defaultValue, &error.outPtr()))
+            g_warning("Failed to register setting %s: %s", setting.key, error->message);
+    }
```

Comment:
I think these should be registered by wpe platform, not by WebKit. Because they are read by WebKit, but they are supposed to be written by the platforms. For example, a GTK4 platform would write them from GtkSettings.